API key, secure connections, and further small changes to Elastic connection wrapper

filipecosta90 opened the pull request:
This PR includes changes aimed at enhancing the Elastic connection wrapper. 
- It updates environment variables for configuration, improves the Elasticsearch client initialization with new parameters like timeout settings, and revises the handling of secure connections. 
- Additionally, it cleans up the code by removing redundant configurations and streamlining the process for setting up Elastic clients across different modules.

filipecosta90 committed:
Enabled api key elastic connections

filipecosta90 committed:
Fixes per pre-commit hook

filipecosta90 committed:
popping the parallel config from a deep copy of search_params

KShivendu commented on engine/clients/elasticsearch/config.py:
```diff
@@ -1,4 +1,40 @@
+
+def get_es_client(host, connection_params):
+    client: Elasticsearch = None
+    init_params = {
+        **{
+            "verify_certs": False,
+            "request_timeout": ELASTIC_TIMEOUT,
+            "retry_on_timeout": True,
+        },
+        **connection_params,
+    }
```
```suggestion
    init_params = {
        "verify_certs": False,
        "request_timeout": ELASTIC_TIMEOUT,
        "retry_on_timeout": True,
        **connection_params,
    }
```

KShivendu commented on engine/clients/elasticsearch/config.py:
```diff
@@ -1,4 +1,40 @@
+    else:
+        url = "http://"
+    url += f"{host}:{ELASTIC_PORT}"
+    if ELASTIC_API_KEY is None:
+        client = Elasticsearch(
+            url,
+            basic_auth=(ELASTIC_USER, ELASTIC_PASSWORD),
+            **init_params,
+        )
+    else:
+        client = Elasticsearch(
+            url,
+            api_key=ELASTIC_API_KEY,
+            **init_params,
+        )
+    return client
```
```suggestion
    if ELASTIC_API_KEY is None:
        return Elasticsearch(
            url,
            basic_auth=(ELASTIC_USER, ELASTIC_PASSWORD),
            **init_params,
        )
    else:
        return Elasticsearch(
            url,
            api_key=ELASTIC_API_KEY,
            **init_params,
        )
```

KShivendu commented on engine/clients/elasticsearch/config.py:
```diff
@@ -1,4 +1,40 @@
+
+
+def get_es_client(host, connection_params):
+    client: Elasticsearch = None
```
```suggestion
```

KShivendu reviewed (changes requested):
Thanks again for the PR! However, I'm not sure if we want to bother about auth and API keys because we don't benchmark cloud deployments. Will think about this.

filipecosta90 committed:
Merge remote-tracking branch 'qdrant/master' into elastic.cloud

filipecosta90 committed:
Update engine/clients/elasticsearch/config.py

Co-authored-by: Kumar Shivendu <kshivendu1@gmail.com>

filipecosta90 committed:
Update engine/clients/elasticsearch/config.py

Co-authored-by: Kumar Shivendu <kshivendu1@gmail.com>

filipecosta90 committed:
Update engine/clients/elasticsearch/config.py

Co-authored-by: Kumar Shivendu <kshivendu1@gmail.com>

filipecosta90 committed:
Removed api_key usage per review

filipecosta90 committed:
Enable specifying the elastic index timeouts. disabled ssl warnings

filipecosta90 commented:
> Thanks again for the PR! However, I'm not sure if we want to bother about auth and API keys because we don't benchmark cloud deployments. Will think about this.

@KShivendu I've removed the API key initialization. I've also extended the configurations that can be tuned and added extra checks concerning the index status and client connection. Please revise this current version

KShivendu commented:
Hi @filipecosta90, I've implemented your refactor in a slightly cleaner way https://github.com/qdrant/vector-db-benchmark/pull/120

Also, I'm not sure if it makes sense to retry forcing ES into using 1 segment 30 times. Is there a simpler way to do this?

filipecosta90 commented:
> Also, I'm not sure if it makes sense to retry forcing ES into using 1 segment 30 times. Is there a simpler way to do this?

we can wrap it around a backoff. but lets close this PR, merge your changes and then if needed I can open a 2nd iteration PR with further changes.
